### PR TITLE
[risk=low][RW-8325] Update cloudresourcemanager again

### PR DESCRIPTION
**Pull request by jmthibault79**
Repeat #6814 but update the code so Controlled Tier Workspace Creation/Duplication still works.

Tested locally by creating and duplicating CT workspaces

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/google/CloudResourceManagerServiceImpl.java [hunk 1]
```diff
@@ -111,7 +111,7 @@ public Policy getIamPolicy(String googleProject) {
           return serviceCloudResouceManager
               .get()
               .projects()
-              .getIamPolicy(googleProject, new GetIamPolicyRequest())
+              .getIamPolicy("projects/" + googleProject, new GetIamPolicyRequest())
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/google/CloudResourceManagerServiceImpl.java:114]: Here is the fix (see also line 127): adding "projects/"

File: api/src/main/java/org/pmiops/workbench/api/CloudTaskUserController.java [hunk 2]
```diff
@@ -79,17 +78,17 @@ public ResponseEntity<Void> auditProjectAccess(AuditProjectAccessRequest request
                 .filter(
                     project ->
                         project.getParent() == null
-                            || !(ALLOWED_PARENT_IDS.contains(project.getParent().getId())))
+                            || !(ALLOWED_PARENT_IDS.contains(getId(project.getParent()))))
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/CloudTaskUserController.java:81]: getParent() returns a string in the new API.  Previously it was a ResourceType (I think?) which had getType() and getId() accessors.

File: api/src/main/java/org/pmiops/workbench/google/CloudResourceManagerServiceImpl.java [hunk 3]
```diff
@@ -37,7 +37,7 @@ public class CloudResourceManagerServiceImpl implements CloudResourceManagerServ
   public static final String ADMIN_SERVICE_ACCOUNT_NAME = "cloud-resource-admin";
 
   public static final List<String> SCOPES =
-      Arrays.asList(CloudResourceManagerScopes.CLOUD_PLATFORM_READ_ONLY);
+      Collections.singletonList(CloudResourceManagerScopes.CLOUD_PLATFORM);
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/google/CloudResourceManagerServiceImpl.java:40]: The new API requires a stronger scope.  We had to update the permissions on some cloud users accordingly: https://precisionmedicineinitiative.atlassian.net/browse/PD-7677

**Review by evrii — APPROVED** [review]
Tested locally and it worked.

I do not quite understand the scoping changes though.


https://user-images.githubusercontent.com/24514630/180517191-ff82a270-afc3-4f41-a304-0de517db915d.mov

**Comment by jmthibault79** [comment]
I don't know why higher scopes are needed in the new API, but they are

**Pull request merged**